### PR TITLE
feat: refactor action

### PR DESCRIPTION
**Pull request by celiogafesi**
This PR refactors this action to work together with the [`actions/setup-java`](https://github.com/Tradeshift/actions-setup-java) and [`tradeshift/actions-setup-maven`](https://github.com/Tradeshift/actions-setup-maven) actions to replace the current setup coded into the [`tradeshift/actions-setup-java`](https://github.com/Tradeshift/actions-setup-java) fork, setting up java and maven with the necessary certficates.

The change will allow us to avoid having a diverging fork from the [`actions/setup-java`](https://github.com/Tradeshift/actions-setup-java) official repo, which currently makes it hard to pull changes from upstream with new features and/or security patches.

A working example of the new setup can be seen [here](https://github.com/Tradeshift/tradeshift-maven-3rdparty/pull/67).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,71 +1,57 @@
...
-          MAVEN_P12_PASSWORD: ${{ secrets.MAVEN_P12_PASSWORD }}
-          ROOTCA_B64: ${{ secrets.CACERT }}
-
+  - uses: tradeshift/actions-setup-java-mtls@v1
+    with:
+      maven-settings: ${{ secrets.MAVEN_SETTINGS }}
+      maven-security-settings: ${{ secrets.MAVEN_SECURITY }}
+      maven-p12-keystore: ${{ secrets.MAVEN_P12 }}
+      maven-p12-keystore-password: ${{ secrets.MAVEN_P12_PASSWORD }}
+      company-rootca: ${{ secrets.MTLS_CACERT }}
```
Comment (wejendorp) [inline README.md:14]: The CA arg doesn't match the action input name below.

File: README.md [hunk 2]
```diff
@@ -1,71 +1,57 @@
...
-        uses: tradeshift/actions-maven@v1
+      - name: Setup java mlts and maven settings
+        uses: tradeshift/actions-setup-java-mtls@v1
         with:
           maven-settings: ${{ secrets.MAVEN_SETTINGS }}
           maven-security-settings: ${{ secrets.MAVEN_SECURITY }}
           maven-p12-keystore: ${{ secrets.MAVEN_P12 }}
           maven-p12-keystore-password: ${{ secrets.MAVEN_P12_PASSWORD }}
-          company-rootca: ${{ secrets.CACERT }}
+          company-rootca: ${{ secrets.MTLS_CACERT }}
```
Comment (wejendorp) [inline README.md:42]: Same here, wrong arg name.

Comment (wejendorp) [reply README.md:42]: Actually all of the ones we renamed in `action.yml` seem wrong in the docs.

Comment (celiogafesi) [reply README.md]: fixed!

**Pull request merged**